vm/devices/support/fs/lxutil/src/windows/symlink.rs

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use pal::windows::UnicodeString;
+use widestring::U16CString;
+use windows::Wdk::Storage::FileSystem;
+use windows::Win32::System::SystemServices as W32Ss;
+
+/// Get the symlink substitute name and flags from the reparse data.
+fn get_substitute_name(
+    reparse: &FileSystem::REPARSE_DATA_BUFFER,
+) -> lx::Result<(UnicodeString, u32)> {
```

Comment:
Seems like this could return `UnicodeStringRef<'_>` instead of an owning `UnicodeString`.